Common/src/Storage/SessionTableExtensions.cs

```diff
@@ -110,4 +112,169 @@ public static class SessionTableExtensions
                                          e);
     }
   }
+
+  /// <summary>
+  ///   Cancel a session
+  /// </summary>
+  /// <param name="sessionTable">Interface to manage sessions lifecycle</param>
+  /// <param name="sessionId">Id of the session to cancel</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  ///   The metadata of the cancelled session
+  /// </returns>
+  public static async Task<SessionData> CancelSessionAsync(this ISessionTable sessionTable,
+                                                           string             sessionId,
+                                                           CancellationToken  cancellationToken = default)
+    => await sessionTable.UpdateOneSessionAsync(sessionId,
+                                                data => data.Status == SessionStatus.Running || data.Status == SessionStatus.Paused ||
+                                                        data.Status != SessionStatus.Cancelled,
```

Comment:
Your condition is equivalent to the following: 
```c#
data => data.Status != SessionStatus.Cancelled
```

So I think it is not what you want.
```suggestion
                                                data => data.Status == SessionStatus.Running || data.Status == SessionStatus.Paused,
```

Reply:
That's equivalent to 

```csharp
data => data.Status != SessionStatus.Cancelled
```

only when there is the three statuses.

I don't want to cancel purged or deleted sessions.

Reply:
> I don't want to cancel purged or deleted sessions.

That's my point. Currently, you are able to cancel purged and deleted sessions. Your condition is all ORs, so `data.Status == SessionStatus.Paused` is included in `data.Status != SessionStatus.Cancelled`

Reply:
Yeah! You are right. I changed the conditions and added some tests.